src/app/dynamic-forms/models/cinchy-form.model.ts

```diff
@@ -848,7 +928,269 @@ export class Form implements IForm {
+            const valueArray = (isNullOrUndefined(Rowelement[field.cinchyColumn.name]) ?
+              [] :
+              Rowelement[field.cinchyColumn.name].split(","));
+
+            let optionArray = [];
+
+            valueArray.forEach((element: any) => {
+
+              if (element !== "" && !isNullOrUndefined(element)) {
+                const value = element.trim();
+                const objValues = {
+                  id: value,
+                  itemName: value
+                };
+                optionArray.push(objValues);
+              }
+            });
+
+            field.setInitialValue(optionArray);
+          } else {
+            field.setInitialValue(Rowelement[field.cinchyColumn.name]);
+          }
+
+          if (field.cinchyColumn.dataType === "Link") {
+            if (!isNullOrUndefined(Rowelement[field.cinchyColumn.name]) && Rowelement[field.cinchyColumn.name] !== "") {
+              let optionArray: DropdownOption[] = [];
+              const labelForColumn = field.cinchyColumn.isDisplayColumn ? field.cinchyColumn.linkTargetColumnName : field.cinchyColumn.name;
+
+              optionArray.push(new DropdownOption(Rowelement[field.cinchyColumn.name], Rowelement[labelForColumn + " label"]));
+
+              let result = new DropdownDataset(optionArray);
+
+              if (isNullOrUndefined(field.dropdownDataset)) {
+                field.dropdownDataset = result;
+              } else {
+                if (!isNullOrUndefined(field.dropdownDataset.options)) {
+                  field.dropdownDataset.options.push(result.options[0]);
+                }
+              }
+
+              if (!isNullOrUndefined(field.dropdownDataset)) {
+                let dropdownResult = field.dropdownDataset.options.find(e => e.id === Rowelement[field.cinchyColumn.name]);
+                if (!isNullOrUndefined(dropdownResult)) {
+                  if (!field.cinchyColumn.isDisplayColumn) {
+                    Rowelement[field.cinchyColumn.name] = dropdownResult["label"];
+                  }
+                }
+
+              }
+            }
+
+            if (!field.cinchyColumn.isDisplayColumn) {
+              delete Rowelement[field.cinchyColumn.name + " label"];
+            }
+          }
+        });
+      });
+
+      section.flattenedChildFormRecordValues = rowData;
+    });
+
+    // TODO: determine if this is relevant or necessary
+    this.rowId = rowId;
+  }
+
+
+  /**
+   * Generates sections for this form based on the given metadata
+   */
+  populateSectionsFromFormMetadata(metadata: Array<IFormSectionMetadata>): void {
+
+    this._sections = metadata.map(_ => {
+
+      let result = new FormSection(_.id, _.name);
+
+      result.columnsInRow = _.columnsInRow;
+      result.autoExpand = _.autoExpand;
+
+      return result;
+    });
+  }
+
+
+  /**
+   * Ensures that every field's form value correctly points to this form
+   */
+  restoreFormReferenceOnAllFields(): void {
```

Comment:
In what scenario would the fields reference another form? or is this an unknown scenario and we added this to bandage up that occurrence?